plugins/module_utils/prerequisites.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Jan Kodera (jkodera@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import sys
+import re
+import importlib
+import importlib.metadata
+import json
+
+# This file exists because Ansible currently cannot declare dependencies on Python modules
+# when imported. When imported, it fails with Ansible-compatible message if requirements
+# listed below are not installed
+# see https://github.com/ansible/ansible/issues/62733 for more info about lack of req support
+
+#############################
+
+# 'module_name, package_name, version_requirements' triplets
+DEPENDENCIES = [
+    ("fusion", "purefusion", ">=1.0.11,<2.0")
```

Comment:
If the functionality is the same, I'd suggest using `urllib` as it is part of the [standard python library](https://docs.python.org/3.8/library/)

Reply:
Urllib3 is always present if purefusion is present since it's a dependency, so it should be handled by pip dependency management. I'll add it anyways so we can be sure. @andreepyro urllib3 is a dependency of our python SDK, so we would have to modify the generator to change it to std urllib. We're probably going to have to accept it as dependency for now